supertokens_python/recipe/session/session_class.py

```diff
@@ -112,3 +119,77 @@ async def get_expiry(self, user_context: Union[Dict[str, Any], None] = None) ->
             raise_unauthorised_exception("Session does not exist anymore.")
 
         return session_info.expiry
+
+    async def assert_claims(
+        self,
+        claim_validators: List[SessionClaimValidator],
+        user_context: Union[Dict[str, Any], None] = None,
+    ) -> None:
+        if user_context is None:
+            user_context = {}
+
+        original_session_claim_payload_json = json.dumps(
+            self.get_access_token_payload()
+        )
+
+        new_access_token_payload = await update_claims_in_payload_if_needed(
+            claim_validators,
+            self.get_access_token_payload(),
+            self.get_user_id(),
+            user_context,
+        )
+
+        if json.dumps(new_access_token_payload) != original_session_claim_payload_json:
+            await self.merge_into_access_token_payload(
+                new_access_token_payload, user_context
+            )
+
+        validation_errors = await validate_claims_in_payload(
+            claim_validators, new_access_token_payload, user_context
+        )
+
+        if len(validation_errors) > 0:
+            raise_invalid_claims_exception("INVALID_CLAIMS", validation_errors)
+
+    async def fetch_and_set_claim(
+        self, claim: SessionClaim[Any], user_context: Union[Dict[str, Any], None] = None
+    ) -> None:
+        update = await claim.build(self.get_user_id(), user_context)
+        return await self.merge_into_access_token_payload(update, user_context)
+
+    async def set_claim_value(
+        self,
+        claim: SessionClaim[_T],
+        value: _T,
+        user_context: Union[Dict[str, Any], None] = None,
+    ) -> None:
+        update = claim.add_to_payload_({}, value, user_context)
+        return await self.merge_into_access_token_payload(update, user_context)
+
+    async def get_claim_value(
+        self, claim: SessionClaim[Any], user_context: Union[Dict[str, Any], None] = None
+    ) -> Union[Any, None]:
+        return claim.get_value_from_payload(
+            self.get_access_token_payload(), user_context
+        )
+
+    async def remove_claim(
+        self, claim: SessionClaim[Any], user_context: Union[Dict[str, Any], None] = None
+    ) -> None:
+        if user_context is None:
+            user_context = {}
+        update = claim.remove_from_payload_by_merge_({}, user_context)
+        return await self.merge_into_access_token_payload(update, user_context)
+
+    async def merge_into_access_token_payload(
+        self, access_token_payload_update: Dict[str, Any], user_context: Any
+    ) -> None:
+        update_payload = {
+            **self.get_access_token_payload(),
+            **access_token_payload_update,
+        }
+        for k in access_token_payload_update.keys():
+            if access_token_payload_update[k] is None:
+                del update_payload[k]
```

Comment:
See my comment about this in the recipe impl function for merge_into_access_token_payload

Reply:
TODO

Reply:
Asumming you're talking about this:
> if k does not exist, would if new_access_token_payload[k] is None: be true?

The answer remains the same:

It's impossible that k won't exist in new_access_token_payload because

```python
new_access_token_payload = {
      **session_info.access_token_payload,
     **access_token_payload_update,
}
# this merges the two dictionaries
```

but even assuming that this magically happens, python will throw error if k doesn't exist. So the condition can't be true.